src/c++/perf_analyzer/concurrency_worker.cc

```diff
@@ -263,6 +258,10 @@ ConcurrencyWorker::GetCtxId()
     // Find the next available context id to use for this request
     {
       std::lock_guard<std::mutex> lk(cb_mtx_);
+
+      if (free_ctx_ids_.size() < 1) {
```

Comment:
This was a seg fault that was being hit during my triage, related to a race condition where we would try to issue a request before the context had been put back on the free_ctx_ids list. I think the above change to when/where total_ongoing_requests_ was decremented was the fix to avoid hitting this assert. I felt it was safer to keep the assert in just so we don't straight segfault in the future.